onnxoptimizer/passes/eliminate_duplicate_initializer.h

```diff
@@ -0,0 +1,153 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+#pragma once
+
+// Before:
+//   A, B are in the initializer list, and A is equal to B
+//   E = Add(D, A)
+//   F = Add(F, B)
+//   G = Add(E, F)
+// After:
+//   A is in the initializer list
+//   E = Add(D, A)
+//   F = Add(F, A)
+//   G = Add(E, F)
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateDuplicateInitializer final : public FullGraphBasedPass {
+  explicit EliminateDuplicateInitializer()
+      : FullGraphBasedPass(
+            PassType::Nop,
+            PassEfficiency::Complete,
+            PassOptimizationType::Compute) {}
```

Comment:
Should be `PassOptimizationType::Memory`?

Reply:
Fixed.